preprocessor.py

```diff
@@ -0,0 +1,133 @@
+import re
+import sys
+
+class PreProcessor:
+  def __init__(self,url_repl='', email_repl='', phone_repl='', num_repl='', date_repl='', emoji_repl='', html_tag_repl=''):
+    self.url_repl = url_repl
+    self.email_repl = email_repl
+    self.phone_repl = phone_repl
+    self.date_repl = date_repl
+    self.num_repl = num_repl
+    self.emoji_repl = emoji_repl
+    self.html_tag_repl = html_tag_repl
+
+    self._preProccess =[]
+    self._init_preProccess()
+    
+  def _preprocess_email(self,input_text,email_repl):
+    regex_pattern = re.compile(
+        r"(?:mailto:)?"
+        r"(?:^|(?<=[^\w@.)]))([\w+-](\.(?!\.))?)*?[\w+-]@(?:\w-?)*?\w+(\.([a-z]{2,})){1,3}"
+        r"(?:$|(?=\b))",
+        flags=re.IGNORECASE,)
+    text= regex_pattern.sub(email_repl,input_text)
+    return text
+
+  def _preprocess_phone_numbers(self,input_text,phone_repl):
+    # for more information: https://github.com/jfilter/clean-text/issues/10
+    regex_pattern = re.compile(r"((?:\+\d{2}[-\.\s]??|\d{4}[-\.\s]??)?(?:\d{3}[-\.\s]??\d{3}[-\.\s]??\d{4}|\(\d{3}\)\s*\d{3}[-\.\s]??\d{4}|\d{3}[-\.\s]??\d{4}))")
+    #regex_pattern= re.compile(r"((?:^|(?<=[^\w)]))(((\+?[01])|(\+\d{2}))[ .-]?)?(\(?\d{3,4}\)?/?[ .-]?)?(\d{3}[ .-]?\d{4})(\s?(?:ext\.?|[#x-])\s?\d{2,6})?(?:$|(?=\W)))|\+?\d{4,5}[ .-/]\d{6,9}")
+    text = regex_pattern.sub(phone_repl,input_text)
+    return text
+ 
+  def _preprocess_dates(self,input_text,date_repl): 
+    regex_pattern = re.compile(r"(\d{4}(.|-|\/)(0[1-9]|1[0-2])(.|-|\/)(0[1-9]|[12][0-9]|3[01]))|(([1-9]|1[0-9]|2[0-9]|3[0-1]|0[0-9])(.|-|\/)([1-9]|1[0-2]|0[0-9])(.|-|\/)(\d{4}))")
+    text = regex_pattern.sub(date_repl,input_text)
+    return text
+
+  def _preprocess_numbers(self,input_text,num_repl):
+    regex_pattern = re.compile(r"(?:^|(?<=[^\w,.]))[+–-]?(([1-9]\d{0,2}(,\d{3})+(\.\d*)?)|([1-9]\d{0,2}([ .]\d{3})+(,\d*)?)|(\d*?[.,]\d+)|\d+)(?:$|(?=\b))")
+    text= regex_pattern.sub(num_repl,input_text)
+    return text
+
+  def _preprocess_emoji(self,input_text,emoji_repl):
+    # https://gist.github.com/Alex-Just/e86110836f3f93fe7932290526529cd1
+    regex_pattern = re.compile(
+        "(["
+        "\U0001F1E0-\U0001F1FF"  # flags (iOS)
+        "\U0001F300-\U0001F5FF"  # symbols & pictographs
+        "\U0001F600-\U0001F64F"  # emoticons
+        "\U0001F680-\U0001F6FF"  # transport & map symbols
+        "\U0001F700-\U0001F77F"  # alchemical symbols
+        "\U0001F780-\U0001F7FF"  # Geometric Shapes Extended
+        "\U0001F800-\U0001F8FF"  # Supplemental Arrows-C
+        "\U0001F900-\U0001F9FF"  # Supplemental Symbols and Pictographs
+        "\U0001FA00-\U0001FA6F"  # Chess Symbols
+        "\U0001FA70-\U0001FAFF"  # Symbols and Pictographs Extended-A
+        "\U00002702-\U000027B0"  # Dingbats
+        "])"
+      )
+    text= regex_pattern.sub(emoji_repl,input_text)
+    return text
+
+  def _preprocess_html_tags(self,input_text,html_tag_repl):
+    regex_pattern = re.compile(r"<[^>]*>")
```

Comment:
* this is going to remove everyting between < > even if were to be Sinhala words right?